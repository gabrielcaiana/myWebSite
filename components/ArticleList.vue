<template>
  <Container>
    <ul class="max-w-5xl mx-auto">
      <li
        v-for="article of articles"
        :key="article.slug"
        class="mb-12 bg-white bg-opacity-5 p-5 rounded shadow"
      >
        <NuxtLink
          :to="{ name: 'articles-slug', params: { slug: article.slug } }"
          class="md:grid md:gap-4 md:grid-cols-2 flex items-center"
        >
          <img
            v-if="article.img"
            :src="article.img"
            alt=""
            class="mb-4 border rounded"
          />
          <div>
            <h2 class="font-bold text-2xl mb-2">
              {{ article.title }}
            </h2>
            <p class="text-lg">{{ article.description }}</p>
            <p
              class="
                font-bold
                text-blue-500
                hover:text-blue-700
                transition
                duration-300
                ease-in-out
                mt-2
              "
            >
              Read more<span
                class="
                  text-blue-500
                  hover:text-blue-700
                  transition
                  duration-300
                  ease-in-out
                "
                >&hellip;</span
              >
            </p>
          </div>
        </NuxtLink>
      </li>
    </ul>
    <div v-if="total" class="constainer mx-auto my-5 max-w-5xl">
      <Pagination v-if="total > 4" :total="total" />
    </div>
  </Container>
</template>

<script>
export default {
  name: 'ArticleList',
  props: {
    articles: {
      type: Array,
      default: Array,
    },
    total: {
      type: Number,
      default: 0,
    },
  },
}
</script>
