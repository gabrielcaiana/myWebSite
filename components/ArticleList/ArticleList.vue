<template>
  <div>
    <ul class="flex flex-wrap gap-4">
      <li
        v-for="article of articles"
        :key="article.slug"
        class="w-[300px] md:w-[416px] bg-primary-700 cursor-pointer transform transition duration-500 hover:scale-105"
      >
        <NuxtLink :to="`/articles/${article.slug}`">
          <div>
            <img
              class="w-full h-64 object-cover mb-4"
              :src="article.img"
              :alt="article.title"
            />
            <div class="px-4 my-4">
              <h2 class="font-bold text-xl mb-2 text-white font-general-medium">
                {{ article.title }}
              </h2>
              <p
                class="text-sm line-clamp-4 text-primary-200 font-general-regular"
              >
                {{ article.description }}
              </p>
            </div>
          </div>
        </NuxtLink>
      </li>
    </ul>
    <div v-if="total" class="constainer mx-auto my-5 max-w-5xl">
      <Pagination v-if="total > 4" :total="total" />
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'ArticleList',
  props: {
    articles: {
      type: Array,
      default: Array,
    },
    total: {
      type: Number,
      default: 0,
    },
  },
})
</script>
