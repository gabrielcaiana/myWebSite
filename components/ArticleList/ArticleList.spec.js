import { mount } from '@vue/test-utils'
import ArticleList from '~/components/ArticleList/ArticleList.vue'

describe('ArticleList', () => {
  let wrapper

  beforeAll(() => {
    wrapper = mount(ArticleList, {})
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('create Snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
