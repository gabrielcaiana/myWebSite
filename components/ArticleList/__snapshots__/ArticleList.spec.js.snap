// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList create Snapshot 1`] = `
<container>
  <ul
    class="max-w-5xl mx-auto"
  />
   
  <!---->
</container>
`;
